mirrord/agent/src/steal/ip_tables/redirect.rs

```diff
@@ -98,6 +98,83 @@ where
     }
 }
 
+pub struct StandardRedirect<IPT: IPTables> {
+    preroute: PreroutingRedirect<IPT>,
+}
+
+impl<IPT> StandardRedirect<IPT>
+where
+    IPT: IPTables,
+{
+    const ENTRYPOINT: &'static str = "OUTPUT";
+
+    pub fn create(ipt: Arc<IPT>) -> Result<Self> {
+        let preroute = PreroutingRedirect::create(ipt)?;
+
+        Ok(StandardRedirect { preroute })
+    }
+
+    pub fn load(ipt: Arc<IPT>) -> Result<Self> {
+        let preroute = PreroutingRedirect::load(ipt)?;
+
+        Ok(StandardRedirect { preroute })
+    }
+}
+
+/// This wrapper adds a new rule to the NAT OUTPUT chain to redirect "localhost" traffic as well
+/// Note: OUTPUT chain is only traversed for packets produced by local applications
+#[async_trait]
+impl<IPT> Redirect for StandardRedirect<IPT>
+where
+    IPT: IPTables + Send + Sync,
+{
+    async fn mount_entrypoint(&self) -> Result<()> {
+        self.preroute.managed.inner().add_rule(
+            PreroutingRedirect::<IPT>::ENTRYPOINT,
+            &format!("-j {}", self.preroute.managed.chain_name()),
+        )?;
+
+        self.preroute.managed.inner().add_rule(
+            Self::ENTRYPOINT,
+            &format!("-j {}", self.preroute.managed.chain_name()),
+        )?;
+
+        Ok(())
+    }
+
+    async fn unmount_entrypoint(&self) -> Result<()> {
+        self.preroute.managed.inner().remove_rule(
+            PreroutingRedirect::<IPT>::ENTRYPOINT,
+            &format!("-j {}", self.preroute.managed.chain_name()),
+        )?;
+
+        self.preroute.managed.inner().remove_rule(
+            Self::ENTRYPOINT,
+            &format!("-j {}", self.preroute.managed.chain_name()),
+        )?;
+
+        Ok(())
+    }
+
+    async fn add_redirect(&self, redirected_port: Port, target_port: Port) -> Result<()> {
+        let redirect_rule =
+            format!("-m tcp -p tcp --dport {redirected_port} -j REDIRECT --to-ports {target_port}");
```

Comment:
I fear just reusing the same chain can be problematic, I would check that a outgoing request to another service listening on the same port as your's does, (I fear the rules would need to be a bit different like adding a filter on gid or something simmilar)